federation/pkg/kubefed/init/init.go

```diff
@@ -583,9 +587,51 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 		},
 	}
 
+	if dnsProviderConfig != "" {
```

Comment:
imho, its ok to segregate the dnsProviderConfig to a separate function, but still we need to do similar mechanism to test out this option (using fake http). @marun, can you suggest any alternative you have in your mind for testing just this function `addDNSProviderConfig`